src/grammar/dml.rs

```diff
@@ -0,0 +1,57 @@
+use super::parse_where_clause;
+use crate::parser::Parser;
+use source_gen::lexer::TokenKind;
+use source_gen::syntax::SyntaxKind;
+use source_gen::T;
+
+#[allow(unused)]
+pub(crate) fn parse_dml(p: &mut Parser) {
+    parse_delete(p);
+}
+
+pub(crate) fn parse_delete(p: &mut Parser) {
+    p.start(SyntaxKind::DeleteStmt);
+    p.expect(T![delete]);
+    p.expect(T![from]);
+    p.expect_one_of(&[T![quoted_ident], T![unquoted_ident]]);
```

Comment:
The table may be schema qualified, such as `hr.emp`. For this we have the `parse_ident` helper function.